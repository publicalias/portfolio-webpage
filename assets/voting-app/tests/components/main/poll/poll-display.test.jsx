"use strict";

//local imports

const PollDisplay = require("../../../../scripts/components/main/poll/poll-display");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { initTestSnapshot, reactTests } = require("test-helpers/react-tests");

//utilities

const { testShallow } = testWrapper(PollDisplay);

const testSnapshot = initTestSnapshot(testShallow);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(PollDisplay));

//poll display

test("poll display should match snapshot", () => testSnapshot());
