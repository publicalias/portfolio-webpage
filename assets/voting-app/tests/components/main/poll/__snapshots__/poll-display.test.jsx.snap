// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poll display should match snapshot 1`] = `
<div
  className="c-poll-display"
>
  <PollTitle
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": 0,
        "polls": Array [],
        "user": Object {},
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={Object {}}
    match={Object {}}
    staticContext={Object {}}
  />
  <hr />
  <PollChart
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": 0,
        "polls": Array [],
        "user": Object {},
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={Object {}}
    match={Object {}}
    staticContext={Object {}}
  />
</div>
`;
